Makefile

```diff
@@ -1,17 +1,77 @@
-.PHONY: all
-all: build
+include $(PWD)/src/toolchain/Makefile
 
-.PHONY: local-release
-local-release:
-	goreleaser release --snapshot --rm-dist
+KEYS := \
+	6B61ECD76088748C70590D55E90A401336C8AAA9 \
+	A8864A8303994E3A18ACD1760CAB4418C834B102 \
+	66039AA59D823C8BD68DB062D3EC673DF9843E7B \
+	DE050A451E6FAF94C677B58B9361DEC647A087BD
 
-.PHONY: test
-test: build/turnkey
-	go test ./...
+ifneq ("$(wildcard $(ROOT)/src/toolchain)","")
+	clone := $(shell git submodule update --init --recursive)
+endif
+
+.DEFAULT_GOAL :=
+.PHONY: default
+default: \
+	toolchain \
+	$(DEFAULT_GOAL) \
+	$(patsubst %,$(KEY_DIR)/%.asc,$(KEYS)) \
+	$(OUT_DIR)/turnkey.linux-x86_64 \
+	$(OUT_DIR)/turnkey.linux-aarch64 \
+	$(OUT_DIR)/turnkey.darwin-x86_64 \
+	$(OUT_DIR)/turnkey.darwin-aarch64 \
+	$(OUT_DIR)/Formula/turnkey.rb \
+	$(OUT_DIR)/release.env \
+	$(OUT_DIR)/manifest.txt
 
-build/turnkey: main.go internal/
-	go build -o build/turnkey main.go
+.PHONY: install
+install: default
+	mkdir -p ~/.local/bin
+	cp $(OUT_DIR)/turnkey.$(HOST_OS)-$(HOST_ARCH) ~/.local/bin/turnkey
+
+.PHONY: test
+test: $(OUT_DIR)/turnkey.linux-x86_64
+	$(call toolchain,' \
+		GOCACHE=/home/build/$(CACHE_DIR) \
+		GOPATH=/home/build/$(CACHE_DIR) \
+		env -C $(SRC_DIR) go test -v ./... \
+	')
 
+# Clean repo back to initial clone state
 .PHONY: clean
-clean:
-	rm -rf dist/ build/
\ No newline at end of file
+clean: toolchain-clean
+	git clean -dfx $(SRC_DIR)
+
+$(KEY_DIR)/%.asc:
+	$(call fetch_pgp_key,$(basename $(notdir $@)))
+
+$(OUT_DIR)/Formula/turnkey.rb: \
```

Comment:
let's pull this make target out for now and bring the formula changes to [homebrew-tap](https://github.com/tkhq/homebrew-tap).

I'll spin up a PR to update the formula in that repo once this PR lands.

Reply:
Let's do this as a followup change